k2/python/k2/autograd.py

```diff
@@ -81,6 +82,110 @@ def backward(ctx, unused: torch.Tensor
             return None, None, None, arc_scores.exp()
 
 
+class _IntersectDensePrunedFunction(torch.autograd.Function):
+
+    @staticmethod
+    def forward(ctx, a_fsas: Fsa, b_fsas: DenseFsaVec, out_fsa: List[Fsa],
+                beam: float, max_active_states: int, min_active_states: int,
+                unused_scores_a: torch.Tensor,
+                unused_scores_b: torch.Tensor) -> torch.Tensor:
+        '''Intersect array of FSAs on CPU/GPU.
+
+        Args:
+          a_fsas:
+            Input FsaVec, i.e., `decoding graphs`, one per sequence. It might
+            just be a linear sequence of phones, or might be something more
+            complicated. Must have either `a_fsas.shape[0] == b_fsas.dim0()`, or
+            `a_fsas.shape[0] == 1` in which case the graph is shared.
+          b_fsas:
+            Input FSAs that correspond to neural network output.
+          out_fsa:
+            A list containing ONLY one entry which will be set to the
+            generated FSA on return. We pass it as a list since the return
+            value can only be types of torch.Tensor in the `forward` function.
+          beam:
+            Decoding beam, e.g. 10.  Smaller is faster, larger is more exact
+            (less pruning). This is the default value; it may be modified by
+            `min_active_states` and `max_active_states`.
+          max_active_states:
+            Maximum number of FSA states that are allowed to be active on any
+            given frame for any given intersection/composition task. This is
+            advisory, in that it will try not to exceed that but may not always
+            succeed.
+          min_active_states:
+            Minimum number of FSA states that are allowed to be active on any
+            given frame for any given intersection/composition task. This is
+            advisory, in that it will try not to have fewer than this number
+            active.
```

Comment:
Say what value to put if there is no constraint.  (Check the C++ code)

Reply:
Added "Set to zero if there is no constraint".